src/main/java/de/hysky/skyblocker/skyblock/dungeon/DungeonMap.java

```diff
@@ -4,48 +4,65 @@
+        ));
+        ClientPlayConnectionEvents.JOIN.register((handler, sender, client) -> reset());
+    }
+
     public static void render(MatrixStack matrices) {
         MinecraftClient client = MinecraftClient.getInstance();
         if (client.player == null || client.world == null) return;
-        ItemStack item = client.player.getInventory().main.get(8);
-        NbtCompound tag = item.getNbt();
-
-        if (tag != null && tag.contains("map")) {
-            String tag2 = tag.asString();
-            tag2 = StringUtils.substringBetween(tag2, "map:", "}");
-            int mapid = Integer.parseInt(tag2);
-            VertexConsumerProvider.Immediate vertices = client.getBufferBuilders().getEffectVertexConsumers();
-            MapRenderer map = client.gameRenderer.getMapRenderer();
-            MapState state = FilledMapItem.getMapState(mapid, client.world);
-            float scaling = SkyblockerConfigManager.get().locations.dungeons.mapScaling;
-            int x = SkyblockerConfigManager.get().locations.dungeons.mapX;
-            int y = SkyblockerConfigManager.get().locations.dungeons.mapY;
-
-            if (state == null) return;
-            matrices.push();
-            matrices.translate(x, y, 0);
-            matrices.scale(scaling, scaling, 0f);
-            map.draw(matrices, vertices, mapid, state, false, 15728880);
-            vertices.draw();
-            matrices.pop();
-        }
+
+        int mapId = getMapId(client.player.getInventory().main.get(8));
+
+        MapState state = FilledMapItem.getMapState(mapId, client.world);
+        if (state == null) return;
+
+        int x = SkyblockerConfigManager.get().locations.dungeons.mapX;
+        int y = SkyblockerConfigManager.get().locations.dungeons.mapY;
+        float scaling = SkyblockerConfigManager.get().locations.dungeons.mapScaling;
+        VertexConsumerProvider.Immediate vertices = client.getBufferBuilders().getEffectVertexConsumers();
+        MapRenderer mapRenderer = client.gameRenderer.getMapRenderer();
+
+        matrices.push();
+        matrices.translate(x, y, 0);
+        matrices.scale(scaling, scaling, 0f);
+        mapRenderer.draw(matrices, vertices, mapId, state, false, LightmapTextureManager.MAX_LIGHT_COORDINATE);
+        vertices.draw();
+        matrices.pop();
     }
 
-	public static void init() { //Todo: consider renaming the command to a more general name since it'll also have dungeon score and maybe other stuff in the future
-        ClientCommandRegistrationCallback.EVENT.register((dispatcher, registryAccess) -> dispatcher.register(ClientCommandManager.literal("skyblocker")
-                .then(ClientCommandManager.literal("hud")
-                        .then(ClientCommandManager.literal("dungeonmap")
-                                .executes(Scheduler.queueOpenScreenCommand(DungeonMapConfigScreen::new))))));
+    public static int getMapId(ItemStack stack) {
+        if (stack.isOf(Items.FILLED_MAP) && stack.hasNbt()) {
```

Comment:
Remove the nbt check as the map id get method already does the same thing, and it will only create more for me to change in the 1.20.5 branch.